### PR TITLE
Add a simple memory eviction test that tests only the admission and node conditions logic

### PR DESCRIPTION
**Pull request by vishh**
cc @mtaufen @derekwaynecarr 

For #31676

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31939)

### CODE REVIEW

**Comment by vishh** [comment]
This test is failing within a minute which is very helpful for debugging.

File: test/e2e_node/memory_qos_eviction_test.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
+/*
```
Comment (mtaufen) [inline test/e2e_node/memory_qos_eviction_test.go]: Might want to wait on #31912 before renaming this file.

**Comment by derekwaynecarr** [comment]
Trying this out now.  Thanks.

File: test/e2e_node/memory_qos_eviction_test.go (hunk 1 above)
Comment (vishh) [inline test/e2e_node/memory_qos_eviction_test.go]: I'm ok with that. I'm going to mark this test flaky for now.

**Comment by derekwaynecarr** [comment]
This test was able to show the repro for me locally, I am digging further.

File: test/e2e_node/memory_eviction_test.go [hunk 2]
```diff
@@ -33,70 +31,49 @@ import (
...
-			}, 60*time.Minute, 5*time.Second).Should(BeNil())
+				return fmt.Errorf("current available memory is: %d bytes. Expected at least %d bytes available.", avail, halflimit)
+			}, 5*time.Minute, 15*time.Second).Should(BeNil())
+		})
+		var guaranteed *api.Pod
+		It("should contain node memory pressure condition)", func() {
+			memoryLimit := getMemoryLimit(f)
+			By("creating a memory hogging pod.")
+			guaranteed = createMemhogPod(f, "guaranteed-", "guaranteed", api.ResourceRequirements{
+				Limits: api.ResourceList{
```
Comment (mtaufen) [inline test/e2e_node/memory_eviction_test.go]: I think that in order to be classed as "guaranteed", you need to also specify `Requests` that are equal to the `Limits`.

Comment (derekwaynecarr) [inline test/e2e_node/memory_eviction_test.go]: Requests will default to Limits if unspecified.

File: test/e2e_node/memory_eviction_test.go [hunk 3]
```diff
@@ -33,70 +31,49 @@ import (
...
+					"cpu": resource.MustParse("100m"),
+					// Set the memory limit to 80% of machine capacity to induce memory pressure.
```
Comment (mtaufen) [inline test/e2e_node/memory_eviction_test.go]: Maybe reword this to "Note: The pod returned by `createMemhogPod` will only consume up to 80% of the memory limit set here."

**Comment by derekwaynecarr** [comment]
Looking at logs run locally:

```
$ cat /tmp/kubelet.log | grep "eviction_manager"
Sep 02 15:09:51 localhost.localdomain kubelet[15100]: W0902 15:09:51.744896   15100 eviction_manager.go:223] eviction manager: attempting to reclaim memory
Sep 02 15:09:51 localhost.localdomain kubelet[15100]: I0902 15:09:51.744928   15100 eviction_manager.go:237] eviction manager: must evict pod(s) to reclaim memory
Sep 02 15:09:51 localhost.localdomain kubelet[15100]: I0902 15:09:51.744950   15100 eviction_manager.go:256] eviction manager: pods ranked for eviction: guaranteed-h9tli_e2e-tests-eviction-test-simple-tg9np(c7c7e36d-7140-11e6-bd65-28d2444e470d)
Sep 02 15:09:53 localhost.localdomain kubelet[15100]: I0902 15:09:53.745150   15100 eviction_manager.go:276] eviction manager: pod guaranteed-h9tli_e2e-tests-eviction-test-simple-tg9np(c7c7e36d-7140-11e6-bd65-28d2444e470d) failed to evict timeout waiting to kill pod
Sep 02 15:09:53 localhost.localdomain kubelet[15100]: I0902 15:09:53.745186   15100 eviction_manager.go:283] eviction manager: unable to evict any pods from the node
Sep 02 15:10:03 localhost.localdomain kubelet[15100]: W0902 15:10:03.788018   15100 eviction_manager.go:223] eviction manager: attempting to reclaim memory
Sep 02 15:10:03 localhost.localdomain kubelet[15100]: I0902 15:10:03.788034   15100 eviction_manager.go:237] eviction manager: must evict pod(s) to reclaim memory
Sep 02 15:10:03 localhost.localdomain kubelet[15100]: E0902 15:10:03.788042   15100 eviction_manager.go:249] eviction manager: eviction thresholds have been met, but no pods are active to evict
Sep 02 15:10:13 localhost.localdomain kubelet[15100]: I0902 15:10:13.833049   15100 eviction_manager.go:216] eviction manager: no resources are starved
Sep 02 15:10:23 localhost.localdomain kubelet[15100]: I0902 15:10:23.839083   15100 eviction_manager.go:216] eviction manager: no resources are starved
```

see the timeout failure...

**Comment by derekwaynecarr** [comment]
With another run (and some more trace):

```
Sep 02 15:20:02 localhost.localdomain kubelet[18530]: I0902 15:20:02.917855   18530 eviction_manager.go:216] eviction manager: no resources are starved
Sep 02 15:20:12 localhost.localdomain kubelet[18530]: W0902 15:20:12.963510   18530 eviction_manager.go:223] eviction manager: attempting to reclaim memory
Sep 02 15:20:12 localhost.localdomain kubelet[18530]: I0902 15:20:12.963531   18530 eviction_manager.go:237] eviction manager: must evict pod(s) to reclaim memory
Sep 02 15:20:12 localhost.localdomain kubelet[18530]: I0902 15:20:12.963548   18530 eviction_manager.go:256] eviction manager: pods ranked for eviction: guaranteed-cn933_e2e-tests-eviction-test-simple-rkrca(33a5f9aa-7142-11e6-a2e9-28d2444e470d)
Sep 02 15:20:14 localhost.localdomain kubelet[18530]: I0902 15:20:14.747890   18530 eviction_manager.go:280] eviction manager: pod guaranteed-cn933_e2e-tests-eviction-test-simple-rkrca(33a5f9aa-7142-11e6-a2e9-28d2444e470d) evicted successfully
Sep 02 15:20:24 localhost.localdomain kubelet[18530]: I0902 15:20:24.777956   18530 eviction_manager.go:216] eviction manager: no resources are starved
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.865967   18530 eviction_manager.go:216] eviction manager: no resources are starved
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931358   18530 eviction_manager.go:98] eviction manager: start Admit: admit-best-effort-pod
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931374   18530 eviction_manager.go:104] eviction manager: has node conditions
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931379   18530 eviction_manager.go:109] eviction manager: has memory pressure
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931388   18530 eviction_manager.go:113] eviction manager: pod qos BestEffort
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931453   18530 eviction_manager.go:120] eviction manager: failing pod
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: W0902 15:20:34.931463   18530 eviction_manager.go:122] Failed to admit pod "admit-best-effort-pod_e2e-tests-eviction-test-simple-hak0n(529ec79f-7142-11e6-a2e9-28d2444e470d)" - node has conditions: [MemoryPressure]
Sep 02 15:20:34 localhost.localdomain kubelet[18530]: I0902 15:20:34.931486   18530 eviction_manager.go:127] eviction manager: end Admit: admit-best-effort-pod
```

**Comment by derekwaynecarr** [comment]
I think I see the problem.

File: test/e2e_node/memory_eviction_test.go [hunk 4]
```diff
@@ -107,48 +84,39 @@ var _ = framework.KubeDescribe("MemoryEviction [Slow] [Serial] [Disruptive]", fu
...
+					return fmt.Errorf("node is not reporting memory pressure condition. Current Conditions: %+v", node.Status.Conditions)
 				}
 				return nil
-			}, 5*time.Minute, 15*time.Second).Should(BeNil())
+			}, 5*time.Minute, 5*time.Second).Should(BeNil())
+		})
+		It("should drop the node memory pressure condition", func() {
+			By("deleting the memory hog pod.")
+			graceperiod := int64(1)
+			err := f.PodClient().Delete(guaranteed.Name, &api.DeleteOptions{GracePeriodSeconds: &graceperiod})
```
Comment (derekwaynecarr) [inline test/e2e_node/memory_eviction_test.go]: we will evict guaranteed pods.  this fails for me locally because the pod is evicted.  we agreed to evict guaranteed pods until we enforce memory limits on kubernetes daemons.

File: test/e2e_node/memory_eviction_test.go (hunk 3 above)
Comment (vishh) [inline test/e2e_node/memory_eviction_test.go]: Ack

File: test/e2e_node/memory_eviction_test.go (hunk 4 above)
Comment (derekwaynecarr) [inline test/e2e_node/memory_eviction_test.go]: the pod client has an incorrect namespace at this point.

after i added some debugging:

INFO: POD CLIENT NAMESPACE: e2e-tests-eviction-test-simple-zvhsk, POD NAMESPACE: e2e-tests-eviction-test-simple-wkoar

so that causes this test to fail for me locally, fixing it up.

Comment (derekwaynecarr) [inline test/e2e_node/memory_eviction_test.go]: basically, from what i can see each It block is getting different namespace, but you are sharing pods across them.

Comment (derekwaynecarr) [inline test/e2e_node/memory_eviction_test.go]: so yeah, as part of ns deletion in prior It block, the pod is deleted already.

File: test/e2e_node/memory_eviction_test.go (hunk 2 above)
Comment (mtaufen) [inline test/e2e_node/memory_eviction_test.go]: Kk good to know.

**Comment by vishh** [comment]
Closing in favor of #32006
